drivers/Serial.h

```diff
@@ -81,6 +81,23 @@ class Serial : public SerialBase, public Stream {
      */
     Serial(PinName tx, PinName rx, int baud);
 
+    /* Stream gives us a FileHandle with non-functional poll()/readable()/writable. Pass through
+     * the calls from the SerialBase instead for backwards compatibility. This problem is
+     * part of why Stream and Serial should be deprecated.
+     */
+    bool readable()
```

Comment:
In practice, I think const-correctness on these sort of device objects doesn't serve that much purpose anyway - no-one is ever going to be passing around `const FileHandle *` or `const Serial *` pointers. There's basically nothing you could do with such a pointer apart from check readability...

I put in the consts when adding those new methods to FileHandle, because why not, but I don't see any pressing reason to try to retrofit const to all the device objects. Serial's readable and writable are currently non-const, via SerialBase, and this preserves that.